### ISSUE TITLE
"bad stack reference in transfer"

### ISSUE THREAD

**Issue by biggeruniverse** [stackless-dev/stackless#218]
No matter what I do, I'm getting title error in Stackless 3.7.3

Very simple example yields this:

`int main(int argc, char *argv[]) {
        Py_Initialize();
	PyRun_SimpleString("def main_main():\n\timport stackless\n\twhile True:\n\t\tstackless.schedule()");
	PyTaskletObject *task = PyTasklet_New(NULL, PyObject_GetAttrString(PyImport_AddModule("__main__"), "main_main"));
	PyTasklet_Setup(task, Py_None, Py_None);
	PyTasklet_Run(task);
        Py_Finalize();
        return 0;
}`

**Comment by akruis** [comment]
Thank you for the report. I can reproduce the problem.

**Comment by akruis** [comment]
A simpler test code is

```
#include <Python.h>
#include <stackless_api.h>

int main(int argc, char *argv[]) {
Py_Initialize(); 
PyRun_SimpleString(
"import stackless\n"
"stackless.tasklet(print)('Hello, World!')\n"
);
if (!PyStackless_Schedule(Py_None, 0))
    PyErr_Print();
Py_Finalize(); 
return 0;
}
```

The bug is caused by commit 441c7d8e5. It adds code in Stackless/core/stacklesseval.c to clear ts->st.cstack_base.

**Comment by akruis** [comment]
I wondered, why test_outside.py did't catch this bug. The reason is simple. `stackless.test_outside()` didn't reset the thread state fully. With a fixed `test_outside()` test_outside.TestOutside.test_outside3 and test_outside.TestOutside.test_outside3_H cause interesting assertion failures.

**Comment by akruis** [comment]
An analysis of test_outside.py showed, that this mechanism can't be used to test embedding of Stackless Python. Therefore I added a specific test module test_slp_embed.py.

And I fixed the bug. Fix will be in 3.7.4-slp
